src/userEvent/touchEvents.ts

```diff
@@ -0,0 +1,115 @@
+import { isHostElementPointerEventEnabled } from '../helpers/isHostElementPointerEventEnabled';
+import { getHostComponentNames } from '../helpers/host-component-names';
+import { jestFakeTimersAreEnabled } from '../helpers/timers';
+import { DEFAULT_MIN_PRESS_DURATION } from './constants';
+import { warnAboutRealTimers } from './utils/warnAboutRealTimers';
+
+const defaultPressEvent = {
+  persist: jest.fn(),
+  nativeEvent: {
+    timestamp: new Date().getTime(),
+  },
+  currentTarget: { measure: jest.fn() },
+};
+
+export type PressOptions = {
+  pressDuration: number;
+};
+
+export const basePress = async (
+  element: ReactTestInstance,
+  options: PressOptions = { pressDuration: 0 }
+): Promise<void> => {
+  // Text and TextInput components are mocked in React Native preset so the mock
+  // doesn't implement the pressability class
+  // Thus we need to call the props directly on the host component
+  const isEnabledHostText =
+    filterNodeByType(element, getHostComponentNames().text) &&
+    !element.props.disabled;
+  const isEnabledTextInput =
+    filterNodeByType(element, getHostComponentNames().textInput) &&
+    element.props.editable !== false;
+
+  if (isEnabledHostText || isEnabledTextInput) {
+    const { onPressIn, onPress, onPressOut } = element.props;
+    if (onPressIn) {
+      onPressIn(defaultPressEvent);
+    }
+    if (onPress) {
+      onPress(defaultPressEvent);
+    }
+    if (onPressOut) {
+      onPressOut(defaultPressEvent);
+    }
+  }
+
+  if (isEnabledTouchResponder(element)) {
+    await triggerPressEvent(element, options);
+    return;
+  }
+
+  const hostParentElement = getHostParent(element);
+  if (!hostParentElement) {
+    return;
+  }
+
+  await basePress(hostParentElement, options);
+};
+
+export const press = async (element: ReactTestInstance): Promise<void> =>
+  basePress(element);
+
+export const longPress = async (
+  element: ReactTestInstance,
+  options: PressOptions = { pressDuration: 500 }
+): Promise<void> => basePress(element, options);
+
+const triggerPressEvent = async (
+  element: ReactTestInstance,
+  options: PressOptions = { pressDuration: 0 }
+) => {
+  const areFakeTimersEnabled = jestFakeTimersAreEnabled();
+  if (!areFakeTimersEnabled) {
+    warnAboutRealTimers();
+  }
+
+  await act(async () => {
+    element.props.onResponderGrant({
+      ...defaultPressEvent,
+      dispatchConfig: { registrationName: 'onResponderGrant' },
+    });
+
+    if (areFakeTimersEnabled) {
+      jest.advanceTimersByTime(options.pressDuration);
+    } else {
+      await wait(options.pressDuration);
+    }
+
+    element.props.onResponderRelease({
+      ...defaultPressEvent,
+      dispatchConfig: { registrationName: 'onResponderRelease' },
+    });
+
+    if (areFakeTimersEnabled) {
+      jest.runOnlyPendingTimers();
+    } else {
+      await wait(DEFAULT_MIN_PRESS_DURATION);
+    }
+  });
+};
+
+const isEnabledTouchResponder = (element: ReactTestInstance) => {
```

Comment:
Q: In the fireEvent we also seem to check `onMoveShouldSetResponder`, does it make sense to include it in this check?

Reply:
This is more related to gestures handling, onMoveShouldSetResponder is used nowhere in the Pressability class, those are two separate things. For now this implementation of press doesn't trigger any gesture events but they should be added in an ulterior version as well